### PR TITLE
Fix utilities include path and add missing headers to meta.yaml

### PR DESCRIPTION
**Pull request by harrism**
#2475 missed updating the type_dispatcher include for one file (not sure how CI passed). This PR: 

- [x] fixes that include
- [x] renames include/cudf/utils to include/cudf/utilities to match other utilities folder names in libcudf.
- [x] Adds all missing includes from the libcudf conda meta.yaml

### CODE REVIEW

File: cpp/src/copying/gather.cu [hunk 1]
```diff
@@ -31,7 +31,7 @@
 #include <cub/cub.cuh>
 #include <nvstrings/NVCategory.h>
 #include <nvstrings/NVStrings.h>
-#include <cudf/utils/legacy/nvcategory_util.hpp>
+#include <cudf/utilities/legacy/nvcategory_util.hpp>
```
Comment (devavret) [inline cpp/src/copying/gather.cu]: File included twice. Probably an old merge issue.

Comment (harrism) [inline cpp/src/copying/gather.cu]:
```suggestion
```

Comment (harrism) [reply cpp/src/copying/gather.cu]: Fixed, thanks.

File: cpp/tests/groupby/hash/single_column_multi_agg.cu [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #include <cudf/legacy/table.hpp>
 #include <tests/utilities/column_wrapper.cuh>
 #include <tests/utilities/compare_column_wrappers.cuh>
+#include <cudf/utilities/legacy/type_dispatcher.hpp>
 #include <cudf/utils/legacy/type_dispatcher.hpp>
```
Comment (harrism) [inline cpp/tests/groupby/hash/single_column_multi_agg.cu]:
```suggestion
```

**Pull request merged**